assets/training/distillation/components/data_generation/src/generate_data.py

```diff
@@ -180,6 +181,14 @@ def get_parser():
         help="This enables Chain of Thought"
     )
 
+    parser.add_argument(
+        "--data_generation_task_type",
+        type=str,
+        required=True,
+        help="This helps in identifying the type of data generation task to generate the synthetic data.",
+        choices=[v.value for v in DataGenerationTaskType]
+    )
+
     return parser
 
 
```

Comment:
Checked the `retry` method & we might be missing exponential backoff. Could we please add?

Reply:
Are we allowed to use this package?
https://pypi.org/project/backoff/

Reply:
We will have to re-release the environment to add this. Might consider to add it in the next iteration.